addons/captives/functions/fnc_doBlindfoldCaptive.sqf

```diff
@@ -69,6 +69,12 @@ if (_state) then { // Blindfold target
 // Handle for things that need to be dropped to the ground
 if (_dropGoggles) then {
     private _weaponHolder = nearestObject [_target, "WeaponHolder"];
+
+    // if _target is in a vehicle, use vehicle inventory as container
+    if ((vehicle _target) != _target) then {
+        _weaponHolder = vehicle _target;
+    };
+
     if (isNull _weaponHolder || {_target distance _weaponHolder > 2}) then {
```

Comment:
I think this distance check could fail on big vics?

Reply:
Does this check distance? Doesn't it just return the vehicle a unit is in?

Reply:
IIRC it'll return model center which could be more than 2m away from the passenger.

Reply:
I'm still confused, isn't all that matters that `_weaponHolder` is assigned a "vehicle" which has an inventory that can be filled, be it a `GroundWeaponHolder` or any other enterable vehicle?

Reply:
There's a distance check on line 78. That's what Pabst mentioned.

`_target distance vehicle _target` might return a number higher than 2 depending on `_target`'s position relative to the center of the vehicle's model, in which case the distance check could fail.

Reply:
`(crew vehicle player) apply {_x distance vehicle _x}`
in the vanilla chinook, last 2 rows (first to load captives) are `3.1073` and `2.63218`

_target = the captive
_weaponHolder = their vehicle

so when `_target distance _weaponHolder > 2` is checked, it will return true
this means it will create a brand-new weapon holder, inside of the vehicle

Reply:
> There's a distance check on line 78. That's what Pabst mentioned.

Ah, yes. I had a feeling that's what was being referred to, but since only my change is highlighted by the review it didn't click.

> so when `_target distance _weaponHolder > 2` is checked, it will return true
> this means it will create a brand-new weapon holder, inside of the vehicle

Got it, when writing the change I thought that the first check at line 78 would prevent that issue, overlooking that it's a `||` 😆

Changing that to `&&` should be sufficient, no?

Reply:
Fixed it.

Reply:
Fix is in review. Changing the condition from OR to AND doesn't handle being more than 2m away from a valid, found weaponHolder.

Reply:
Ah, indeed. Good catch.